frontend/javascripts/oxalis/model/bucket_data_handling/wkstore_adapter.ts

```diff
@@ -167,24 +178,39 @@ export async function requestFromStore(
   const state = Store.getState();
   const isSegmentation = isSegmentationLayer(state.dataset, layerInfo.name);
   const fourBit = state.datasetConfiguration.fourBit && !isSegmentation;
-  const activeMapping = getMappingInfo(
-    state.temporaryConfiguration.activeMappingByLayer,
-    layerInfo.name,
-  );
-  const applyAgglomerates =
-    isSegmentation &&
-    activeMapping != null && // Start to request mapped data during mapping activation phase already
-    activeMapping.mappingStatus !== MappingStatusEnum.DISABLED &&
-    activeMapping.mappingType === "HDF5"
+
+  // Mappings can be applied in the frontend or on the server.
+  const applyAgglomeratesOnServer = (() => {
+    if (!isSegmentation) {
+      return null;
+    }
+    if (needsLocalHdf5Mapping(state, layerInfo.name)) {
+      return null;
+    }
+    const activeMapping = getMappingInfo(
+      state.temporaryConfiguration.activeMappingByLayer,
+      layerInfo.name,
+    );
+    return activeMapping != null && // Start to request mapped data during mapping activation phase already
+      activeMapping.mappingStatus !== MappingStatusEnum.DISABLED &&
+      activeMapping.mappingType === "HDF5"
       ? activeMapping.mappingName
       : null;
+  })();
```

Comment:
Yeah, I think it was named that way because the back-end expects a `applyAgglomerate` parameter that can be a string. I'll change it here to `agglomerateMappingNameToApplyOnServer`.

> You wrapped the logic into a function and directly call this function to encapsule / make the case checking more readable?

Yes :see_no_evil: